opts/opts.go

```diff
@@ -302,6 +302,26 @@ func ValidateSysctl(val string) (string, error) {
 	return "", fmt.Errorf("sysctl '%s' is not whitelisted", val)
 }
 
+// ValidateProxyEnv checks proxy environment variables for validity
+func ValidateProxyEnv(val string) (string, error) {
+	permitted := []string{"HTTP_PROXY", "HTTPS_PROXY", "NO_PROXY", "FTP_PROXY"}
```

Comment:
List for build-args is here; https://github.com/docker/docker/blob/597843e79040a4f00bfe2abd83343d43edd195f5/builder/dockerfile/builder.go#L40-L50